lib/agent/socket_util.js

```diff
@@ -110,13 +111,22 @@ exports.secureSocket = function (socket, host, agent, mock)
     Logger.getInstance().trace('socket reused = %s', socket.isSessionReused());
   };
 
-  // when the socket is secure, perform additional validation
-  socket.on('secure', validate);
+  if (socket.authorized) { // when socket is authorized we can check it now
```

Comment:
What does `authorized` mean in terms of socket? And what do we check? Do you mean validation?

Reply:
rephrased comment, check now